src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1919,6 +1919,56 @@ public sealed partial class DataTableMappingCollection : System.MarshalByRefObje
         System.Data.ITableMapping System.Data.ITableMappingCollection.Add(string sourceTableName, string dataSetTableName) { throw null; }
         System.Data.ITableMapping System.Data.ITableMappingCollection.GetByDataSetTable(string dataSetTableName) { throw null; }
     }
+    public abstract class DbBatch : System.IDisposable, System.IAsyncDisposable
+    {
+        public System.Data.Common.DbBatchCommandCollection BatchCommands { get { throw null; } }
+        protected abstract System.Data.Common.DbBatchCommandCollection DbBatchCommands { get; }
+        public abstract int Timeout { get; set; }
+        public System.Data.Common.DbConnection? Connection { get; set; }
+        protected abstract System.Data.Common.DbConnection? DbConnection { get; set; }
+        public System.Data.Common.DbTransaction? Transaction { get; set; }
+        protected abstract System.Data.Common.DbTransaction? DbTransaction { get; set; }
+        public System.Data.Common.DbDataReader ExecuteReader(System.Data.CommandBehavior behavior = System.Data.CommandBehavior.Default) { throw null; }
+        protected abstract System.Data.Common.DbDataReader ExecuteDbDataReader(System.Data.CommandBehavior behavior);
+        public System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteReaderAsync(System.Threading.CancellationToken cancellationToken = default) { throw null; }
+        public System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteReaderAsync(System.Data.CommandBehavior behavior, System.Threading.CancellationToken cancellationToken = default) { throw null; }
+        protected abstract System.Threading.Tasks.Task<System.Data.Common.DbDataReader> ExecuteDbDataReaderAsync(System.Data.CommandBehavior behavior, System.Threading.CancellationToken cancellationToken);
+        public abstract int ExecuteNonQuery();
+        public abstract System.Threading.Tasks.Task<int> ExecuteNonQueryAsync(System.Threading.CancellationToken cancellationToken = default);
+        public abstract object? ExecuteScalar();
+        public abstract System.Threading.Tasks.Task<object?> ExecuteScalarAsync(System.Threading.CancellationToken cancellationToken = default);
+        public abstract void Prepare();
+        public abstract System.Threading.Tasks.Task PrepareAsync(System.Threading.CancellationToken cancellationToken = default);
+        public abstract void Cancel();
+        public System.Data.Common.DbBatchCommand CreateBatchCommand() { throw null; }
+        protected virtual System.Data.Common.DbBatchCommand CreateDbBatchCommand() { throw null; }
```

Comment:
```suggestion
        protected abstract System.Data.Common.DbBatchCommand CreateDbBatchCommand() { throw null; }
```

Shouldn't it be required for derived classes to implement this?

Reply:
Good catch here too. CreateDbBatch on DbConnection and DbProviderFactory are virtual (and throw NotSupportedException by default) since we can't add abstract methods to existing types without breaking. But DbBatch is a new type, and it makes no sense to define an implementation of it which doesn't support creating a DbBatchCommand.